pkg/workspace/util.go

```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
 		config.Region = region
 	}
 }
+
+// GetIntFieldFromGenericConfig returns the value of the key in config which should be of type int.
+// If exist but not int, return error. If not exist, return 0, nil.
+func GetIntFieldFromGenericConfig(config workspace.GenericConfig, key string) (int, error) {
```

Comment:
compared to return error, return yes or no might be better?

Reply:
If return yes or no, when using these functions, the calling one always still need to build an error. Besides, when calling `GetMapFromGenericConfig`, there are two kinds of errors, the first is `not map`, the second is `field is not string`, using `bool` is not enough.